### ISSUE TITLE
Add property attribute and EdgeQL syntax for collation specification

### ISSUE THREAD

**Issue by elprans** [edgedb/edgedb#386]
We need support for 

`CREATE PROPERTY foo ... { SET collation := 'coll-name' }`

and

`expr COLLATE 'collation'` to be used in comparison and `ORDER BY ` clauses.

**Comment by dimaqq** [comment]
(pardon if this is the wrong place to mention this).
(also, full disclosure, we don't use edgedb, I just think it's awesome)

The system I'm working on, is multi-tenant and multi-language.
I think, in the edgedb setup, we'd keep e.g. user data (incl. name) in a one large table.
And then some customers would query their users with collation X and another with collation Y.
Ideally this would be implemented as several indices (one per locale) over the name "column" so that lookup is efficient.
Collation at run-time may also be fine if it's fast enough, I guess there's always a space-time trade-off...

**Comment by elprans** [comment]
`COLLATE 'collation'` is generic and will be supported in index expressions, so this case should be covered.

**Comment by XeniaLu** [comment]
Is there any update about this feature? 

We have developed a project with a primary focus on Chinese users, and accurate Chinese sorting is absolutely crucial for us, such as sorting users' names. 

This will also have an impact on our future process of switching existing projects from other databases to EdgeDB, as we might have to let go of functionalities related to Chinese sorting.

I really hope that EdgeDB can support this feature.

**Comment by KyleXie** [comment]
yeah, sorting by Chinese characters is widely used in almost all of my projects, it'd be great if we could have this feature in EdgeDB 👍🏻
